src/endpoints/nfts/nft.service.ts

```diff
@@ -66,63 +66,71 @@ export class NftService {
   }
 
   buildElasticNftFilter(filter: NftFilter, identifier?: string, address?: string) {
-    const queries = [];
-    queries.push(QueryType.Exists('identifier'));
+    const mustQueries = [];
+    const shouldQueries = [];
+    mustQueries.push(QueryType.Exists('identifier'));
 
     if (address) {
-      queries.push(QueryType.Match('address', address));
+      mustQueries.push(QueryType.Match('address', address));
     }
 
     if (filter.search !== undefined) {
-      queries.push(QueryType.Wildcard('token', `*${filter.search}*`));
+      mustQueries.push(QueryType.Wildcard('token', `*${filter.search}*`));
     }
 
     if (filter.type !== undefined) {
-      queries.push(QueryType.Match('type', filter.type));
+      mustQueries.push(QueryType.Match('type', filter.type));
     }
 
     if (identifier !== undefined) {
-      queries.push(QueryType.Match('identifier', identifier, QueryOperator.AND));
+      mustQueries.push(QueryType.Match('identifier', identifier, QueryOperator.AND));
     }
 
     if (filter.collection !== undefined && filter.collection !== '') {
-      queries.push(QueryType.Match('token', filter.collection, QueryOperator.AND));
+      mustQueries.push(QueryType.Match('token', filter.collection, QueryOperator.AND));
+    }
+
+    if (filter.collections !== undefined && filter.collections.length !== 0) {
+      for (const collection of filter.collections) {
```

Comment:
you can use a similar approach to how identifiers are handled and then you wouldn't need shouldQueries any more. It will also be easier to extend in the future